core/dbt/deprecations.py

```diff
@@ -43,6 +43,15 @@ def show(self, *args, **kwargs) -> None:
             active_deprecations.add(self.name)
 
 
+class DispatchPackagesDeprecation(DBTDeprecation):
+    _name = 'dispatch-packages'
+    _description = '''\
+    The "packages" argument of adapter.dispatch() has been deprecated. Use the "macro_namespace" argument instead.
+
+    Raised during dispatch for: {macro_name}
+    Read details: https://docs.getdbt.com/reference/dbt-jinja-functions/dispatch
```

Comment:
Nit: It looks like there's a linting error with the deprecation message, so I'll also own that I did not get the whitespace right, haha:
```
* Deprecation Warning: The "packages" argument of adapter.dispatch() has been
deprecated. Use the "macro_namespace" argument instead.
Raised during dispatch for: except Read details:
https://docs.getdbt.com/reference/dbt-jinja-functions/dispatch
```